File: 1. Array/2ndLargest/SecLargest.go
package main

import "fmt"

func main() {
	arr := []int{60, 10, 20, 30, 40, 50, 70}
	fmt.Println(GetSecLargest(arr))
}

var GetSecLargest = func(arr []int) int {
	// TC: O(n)
	// SC: O(n)
	largest_index := GetLargestElement(arr)
	var SecLargestElement int
	if largest_index == 0 {
		SecLargestElement = arr[1]
	} else if largest_index == len(arr)-1 {
		SecLargestElement = arr[len(arr)-2]
	} else {
		SecLargestElement = arr[0]
	}
	for i := 0; i < len(arr); i++ {
		if arr[i] > SecLargestElement && arr[i] < arr[largest_index] {
			SecLargestElement = arr[i]
		}
	}

	return SecLargestElement
}

var GetLargestElement = func(arr []int) int {
	largest_index := 0
	for i := 0; i < len(arr); i++ {
		if arr[i] > arr[largest_index] {
			largest_index = i
		}
	}
	return largest_index
}

File: 1. Array/2ndLargest/SecLargest.java
class SecLargest{
    public static void main(String[] args) {
        int arr[] = new int[]{60, 10, 20, 30, 40, 50, 70};
        System.out.println(FindSecLargest(arr));
    }
    public static int FindSecLargest(int arr[]){
        int largest_index = FindMaximumNumber(arr);
        int secoundLargest = arr[0];
        if(largest_index==0){
            secoundLargest = arr[1];
        }
        else if(largest_index==arr.length-1){
            secoundLargest = arr[arr.length-2];
        }
        else{
            secoundLargest = arr[0];
  
        }
        
        for(int i=0;i<arr.length;i++){
            if(arr[i]>secoundLargest && arr[i]<arr[largest_index]){
                secoundLargest = arr[i];
            }
        }
        return secoundLargest;
    }
    public static int FindMaximumNumber(int arr[]){
        int largest = arr[0];
        int largest_index = 0;
        for(int i=0;i<arr.length;i++){
            if(arr[i]>largest){
                largest = arr[i];
                largest_index = i;
            }
        }
        return largest_index;
    }
}